server/routes/userRoutes.js

```diff
@@ -106,18 +93,56 @@ module.exports = (db) => {
     const { host_name } = req.body.params;
     dbHelper
       .getWebsiteIDByHostname(host_name)
-      // .getWebsiteIDByHostname("reddit.com")
-      .then((site) => dbHelper.addWebsiteToBlacklist(userId, site.id))
+      .then((site) => {
+        if (!site) {
+          // Removes '.com' from end of hostname for DB.
+          const remSuffix = host_name.split(".")[0];
+          // Capitalizes first letter of hostname
+          const name = remSuffix.charAt(0).toUpperCase() + remSuffix.slice(1);
+          const category = null;
+          dbHelper
+          .addWebsite(host_name, name, category)
+          .then((site) => {
+            dbHelper.addWebsiteToBlacklist(userId, site.id);
+          });
+        }
+        dbHelper.addWebsiteToBlacklist(userId, site.id);
+      })
       .catch((err) => res.json(err));
   });
 
-  // Just a test route to test db queries and response
-  // router.get("/test", (req, res) => {
-  //   dbHelper.getQuotaForTodayWithUserID("1").then((quota) => res.json(quota));
-  // });
+  // Need to make a route to remove from user's blacklist
+
+  //   // Just a test route to test db queries and response
+  router.get("/test", (req, res) => {
+    // const { host_name } = req.body.params;
+    const host_name = "google.com";
+    const userId = 1;
+    dbHelper
+      .getWebsiteIDByHostname(host_name)
+      .then((site) => {
+        if (!site) {
+          console.log("=====> Website does not exist yet DB, adding to db");
+          // Removes '.com' from end of hostname for DB.
+          const remSuffix = host_name.split(".")[0];
+          // Capitalizes first letter of hostname
+          const name = remSuffix.charAt(0).toUpperCase() + remSuffix.slice(1);
+          const category = null;
+          dbHelper.addWebsite(host_name, name, category).then((site) => {
+            console.log("=====> Adding a new site to blacklist");
+            console.log(site);
+            dbHelper.addWebsiteToBlacklist(userId, site.id);
+          });
+        }
+        console.log("=====> Website exists in database");
```

Comment:
137-142 also need to be in the `else` block if you want a working test.

Reply:
Fixed on my end